data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/resync/ResyncWorker.java

```diff
@@ -0,0 +1,131 @@
+    private final Buffer<Record<Event>> buffer;
+    private final RecordConverter recordConverter;
+    private final AcknowledgementSet acknowledgementSet;
+
+    ResyncWorker(ResyncPartition resyncPartition,
+                 RdsSourceConfig sourceConfig,
+                 QueryManager queryManager,
+                 Buffer<Record<Event>> buffer,
+                 RecordConverter recordConverter,
+                 AcknowledgementSet acknowledgementSet) {
+        this.resyncPartition = resyncPartition;
+        this.sourceConfig = sourceConfig;
+        this.queryManager = queryManager;
+        this.buffer = buffer;
+        this.recordConverter = recordConverter;
+        this.acknowledgementSet = acknowledgementSet;
+    }
+
+    public static ResyncWorker create(ResyncPartition resyncPartition,
+                                      RdsSourceConfig sourceConfig,
+                                      QueryManager queryManager,
+                                      Buffer<Record<Event>> buffer,
+                                      RecordConverter recordConverter,
+                                      AcknowledgementSet acknowledgementSet) {
+        return new ResyncWorker(resyncPartition, sourceConfig, queryManager, buffer, recordConverter, acknowledgementSet);
+    }
+
+    public void run() {
+        String[] keySplits = resyncPartition.getPartitionKey().split("\\|");
+        final String database = keySplits[0];
+        final String table = keySplits[1];
+        final long eventTimestampMillis = Long.parseLong(keySplits[2]);
+
+        if (resyncPartition.getProgressState().isEmpty()) {
+            final String errorMessage = "ResyncPartition " + resyncPartition.getPartitionKey() + " doesn't contain progress state.";
+            throw new RuntimeException(errorMessage);
+        }
+        final ResyncProgressState progressState = resyncPartition.getProgressState().get();
+        final String foreignKeyName = progressState.getForeignKeyName();
+        final Object updatedValue = progressState.getUpdatedValue();
+
+        LOG.debug("Will perform resync on table: {}.{}, with foreign key name: {}, and updated value: {}", database, table, foreignKeyName, updatedValue);
+        String queryStatement;
+        if (updatedValue == null) {
+            queryStatement = String.format("SELECT * FROM %s WHERE %s IS NULL", database + "." + table, foreignKeyName);
+        } else {
+            queryStatement = String.format("SELECT * FROM %s WHERE %s='%s'", database + "." + table, foreignKeyName, updatedValue);
+        }
+        LOG.debug("Query statement: {}", queryStatement);
+
+        List<Map<String, Object>> rows = queryManager.selectRows(queryStatement);
+        LOG.debug("Found {} rows to resync", rows.size());
+
+        BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, DEFAULT_BUFFER_BATCH_SIZE, BUFFER_TIMEOUT);
+
+        for (Map<String, Object> row : rows) {
```

Comment:
Move rest of the code to new method to process rows

```
processRows(rows, partitionInfo, progressState);
```

Reply:
Done.